diff --git a/node_modules/robotjs/src/keycode.h b/node_modules/robotjs/src/keycode.h
index 9d0b099..f4818e8 100644
--- a/node_modules/robotjs/src/keycode.h
+++ b/node_modules/robotjs/src/keycode.h
@@ -17,6 +17,8 @@ extern "C"
 	
 enum _MMKeyCode {
 	K_NOT_A_KEY = 9999,
+	K_KEY_C = 0x43,
+	K_KEY_V = 0x56,
 	K_BACKSPACE = kVK_Delete,
 	K_DELETE = kVK_ForwardDelete,
 	K_RETURN = kVK_Return,
@@ -114,6 +116,8 @@ typedef CGKeyCode MMKeyCode;
 
 enum _MMKeyCode {
 	K_NOT_A_KEY = 9999,
+	K_KEY_C = 0x43,
+	K_KEY_V = 0x56,
 	K_BACKSPACE = XK_BackSpace,
 	K_DELETE = XK_Delete,
 	K_RETURN = XK_Return,
@@ -208,6 +212,8 @@ typedef KeySym MMKeyCode;
 
 enum _MMKeyCode {
 	K_NOT_A_KEY = 9999,
+	K_KEY_C = 0x43,
+	K_KEY_V = 0x56,
 	K_BACKSPACE = VK_BACK,
 	K_DELETE = VK_DELETE,
 	K_RETURN = VK_RETURN,
diff --git a/node_modules/robotjs/src/keypress.c b/node_modules/robotjs/src/keypress.c
index 2bbdccd..ec19c80 100644
--- a/node_modules/robotjs/src/keypress.c
+++ b/node_modules/robotjs/src/keypress.c
@@ -16,7 +16,7 @@
 /* Convenience wrappers around ugly APIs. */
 #if defined(IS_WINDOWS)
 	#define WIN32_KEY_EVENT_WAIT(key, flags) \
-		(win32KeyEvent(key, flags), Sleep(DEADBEEF_RANDRANGE(63, 125)))
+		(win32KeyEvent(key, flags))
 #elif defined(USE_X11)
 	#define X_KEY_EVENT(display, key, is_press) \
 		(XTestFakeKeyEvent(display, \
@@ -24,8 +24,7 @@
 		                   is_press, CurrentTime), \
 		 XSync(display, false))
 	#define X_KEY_EVENT_WAIT(display, key, is_press) \
-		(X_KEY_EVENT(display, key, is_press), \
-		 microsleep(DEADBEEF_UNIFORM(62.5, 125.0)))
+		(X_KEY_EVENT(display, key, is_press))
 #endif
 
 #if defined(IS_MACOSX)
diff --git a/node_modules/robotjs/src/robotjs.cc b/node_modules/robotjs/src/robotjs.cc
index 8aadd5d..72c2a06 100644
--- a/node_modules/robotjs/src/robotjs.cc
+++ b/node_modules/robotjs/src/robotjs.cc
@@ -288,6 +288,12 @@ struct KeyNames
 
 static KeyNames key_names[] =
 {
+	// https://docs.microsoft.com/en-us/windows/win32/inputdev/virtual-key-codes
+	// http://keycode.info/
+	// https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode#Printable_keys_in_standard_position
+	{ "key_c",          K_KEY_C },
+	{ "key_v",          K_KEY_V },
+
 	{ "backspace",      K_BACKSPACE },
 	{ "delete",         K_DELETE },
 	{ "enter",          K_RETURN },
