diff --git a/node_modules/robotjs/src/keypress.c b/node_modules/robotjs/src/keypress.c
index 2bbdccd..ec19c80 100644
--- a/node_modules/robotjs/src/keypress.c
+++ b/node_modules/robotjs/src/keypress.c
@@ -16,7 +16,7 @@
 /* Convenience wrappers around ugly APIs. */
 #if defined(IS_WINDOWS)
 	#define WIN32_KEY_EVENT_WAIT(key, flags) \
-		(win32KeyEvent(key, flags), Sleep(DEADBEEF_RANDRANGE(63, 125)))
+		(win32KeyEvent(key, flags))
 #elif defined(USE_X11)
 	#define X_KEY_EVENT(display, key, is_press) \
 		(XTestFakeKeyEvent(display, \
@@ -24,8 +24,7 @@
 		                   is_press, CurrentTime), \
 		 XSync(display, false))
 	#define X_KEY_EVENT_WAIT(display, key, is_press) \
-		(X_KEY_EVENT(display, key, is_press), \
-		 microsleep(DEADBEEF_UNIFORM(62.5, 125.0)))
+		(X_KEY_EVENT(display, key, is_press))
 #endif
 
 #if defined(IS_MACOSX)
